### PR TITLE
[Cloud] queue non urgent requests

### LINKED ISSUE #64: [Cloud Engineer] Optimize software and architecture for asynchronous and scheduled jobs
**Describe the pattern you'd like to propose**
Intent: Use efficient software designs and architectures to minimize the average resources required per unit of work. Implement mechanisms that result in even utilization of components to reduce resources that are idle between tasks and minimize the impact of load spikes.

***
**Describe the pattern you'd like to propose**
This pattern involves using efficient software design and architecture so that less resources are required to perform the same tasks. There are different means you can use to achieve this, including:
Queuing non-urgent processing requests
Flattening utilisation across your pipeline with increased serialisation 
Modifying the capacity of some components so that there are no idling resources waiting for input
Optimising the integration of external services with the use of buffers and rate limiting 
Optimising software using the most efficient available hardware 
Maximising utilisation for batch processing with queue-driven architecture, pipeline management and on-demand workers
Scheduling tasks to avoid simultaneous execution of resource contention and load spikes
Scheduling jobs at times of lowest carbon intensity  

**Describe specific emission impact from this pattern**
Using more efficient software design and architecture results in a more even use of components; less resources are left idle and the impact of load spikes is reduced. 
In terms of the SCI impact, this means both M (embodied carbon) and E (energy) are reduced, leading to a lower overall score.

**References to this pattern**
https://docs.aws.amazon.com/wellarchitected/latest/sustainability-pillar/optimize-software-and-architecture-for-asynchronous-and-scheduled-jobs.html

**Additional context**
Add any other context or screenshots about the proposed pattern here.

### PR DESCRIPTION
**Pull request by greenhsu123**
This PR is part of the work required by issue [64](https://github.com/Green-Software-Foundation/green-software-patterns/issues/64): initial commit for queue non-urgent requests

### CODE REVIEW

File: queue-non-urgent-requests.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+## Solution
+
+Utilise a message queue, so requests that do not require immediate processing are queued. 
+
+## SCI Impact
+
+`SCI = (E * I) + M per R`  
+[Sofware Carbon Intensity Spec](https://github.com/Green-Software-Foundation/software_carbon_intensity)
+
+Concerning the SCI equation, optimising peak CPU utilisation will impact two parts:
```
Comment (dubrie) [inline queue-non-urgent-requests.md:26]: I think this should be:
`Concerning the SCI equation, queuing non-urgent requests will impact two parts`

File: queue-non-urgent-requests.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+- `E`: By reducing the total number of computing equipment required, we reduce the total embodied carbon.
```
Comment (dubrie) [inline queue-non-urgent-requests.md:28]: These descriptions should be flipped, E --> Energy, M --> eMbodied emissions

File: queue-non-urgent-requests.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Assumptions
+- There is an assumption that the components in your system can communicate with a message queue asynchronously. If your component is the producer, it can add requests to the queue without waiting for the previous sent to be processed. If your component is the consumer, it will only process requests when they are available. We assume that no components in the system ever stalled while waiting for another. 
+
+
+## Considerations
+- Decoupling components of your system can introduce unnecessary complexity as well as managing a message queue for your system can introduce unnecessary overhead, so careful consideration should be taken to evaluate if a message queue is suited for your system 
+
+
+## References
```
Comment (dubrie) [inline queue-non-urgent-requests.md:39]: There is a similar pattern in pull request, should we link them in their respective Reference sections?

https://github.com/Green-Software-Foundation/green-software-patterns/pull/86

**Review by dubrie — CHANGES_REQUESTED** [review]
Some small updates but overall looks good!

File: queue-non-urgent-requests.md (hunk 3 above)
Comment (greenhsu123) [reply queue-non-urgent-requests.md:39]: I think we should, but should we link a PR though...

File: queue-non-urgent-requests.md (hunk 1 above)
Comment (greenhsu123) [reply queue-non-urgent-requests.md:26]: Good catch! Thanks!

File: queue-non-urgent-requests.md (hunk 2 above)
Comment (greenhsu123) [reply queue-non-urgent-requests.md:28]: Here again, thanks!

File: queue-non-urgent-requests.md (hunk 3 above)
Comment (dubrie) [reply queue-non-urgent-requests.md:40]: agreed

**Comment by dubrie** [comment]
assigning to @dubrie for SME Review

**Review by dubrie — APPROVED** [review]
Made some minor changes for you (typo, link update, and moved to the proper path), but otherwise looks great!

**Pull request merged**